fastify.d.ts

```diff
@@ -526,12 +526,12 @@ declare namespace fastify {
      * internally waits for the .ready() event. The callback is the same as the
      * Node core.
      */
-    listen(callback: (err: Error, address: string) => void): void
-    listen(port: number, callback: (err: Error, address: string) => void): void
-    listen(port: number, address: string, callback: (err: Error, address: string) => void): void
-    listen(port: number, address: string, backlog: number, callback: (err: Error, address: string) => void): void
-    listen(options: ListenOptions, callback: (err: Error, address: string) => void): void
-    listen(sockFile: string, callback: (err: Error, address: string) => void): void
+    listen(callback: (err: Error | null, address: string) => void): void
```

Comment:
I think an overload would be better here:
```ts
listen(callback: (err: Error) => void): void
listen(callback: (err: null, address: string) => void): void
```
If this is supported by TypeScript then please use this format instead of the union operation as discriminant overloads is more secure. Thank you!

Reply:
@dremora have you tried out my suggestion yet?